src/modules/uORB/uORBDevices_nuttx.cpp

```diff
@@ -669,18 +673,12 @@ uORB::DeviceMaster::ioctl(struct file *filp, int cmd, unsigned long arg)
 					*(adv->instance) = group_tries;
 				}
 
-				/* driver wants a permanent copy of the node name, so make one here */
-				objname = strdup(meta->o_name);
-
-				if (objname == nullptr) {
-					return -ENOMEM;
-				}
+				objname = meta->o_name; //no need for a copy, meta->o_name will never be freed or changed
```

Comment:
If the compiler understands that the name is const it should actually be loaded straight from flash - did you change the type and make sure it is? That should in theory kill all allocations of o_name.